src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1693,7 +1698,7 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
 
                         if (success)
```

Comment:
Why don't we want to unconditionally call `ReportUnusedImportsAndFinishCompiling` here?